File: packages/backend-adapters/django/rajutechie_streamkit_django/middleware.py
from django.http import JsonResponse
from .token import verify_token


class RajutechieStreamKitAuthMiddleware:
    def __init__(self, get_response):
        self.get_response = get_response

    def __call__(self, request):
        auth_header = request.META.get("HTTP_AUTHORIZATION", "")
        if auth_header.startswith("Bearer "):
            try:
                token = auth_header[7:]
                request.rajutechie-streamkit_user = verify_token(token)
            except Exception:
                request.rajutechie-streamkit_user = None
        else:
            request.rajutechie-streamkit_user = None

        return self.get_response(request)


def require_rajutechie_streamkit_auth(view_func):
    def wrapper(request, *args, **kwargs):
        if not getattr(request, "rajutechie-streamkit_user", None):
            return JsonResponse({"error": "Authentication required"}, status=401)
        return view_func(request, *args, **kwargs)
    wrapper.__name__ = view_func.__name__
    return wrapper

File: packages/backend-adapters/express/src/middleware.ts
import type { Request, Response, NextFunction } from 'express';
import { RajutechieStreamKitServer } from '@rajutechie-streamkit/server-sdk';

export interface RajutechieStreamKitMiddlewareConfig {
  apiKey: string;
  apiSecret: string;
}

export function streamKitAuth(config: RajutechieStreamKitMiddlewareConfig) {
  const server = new RajutechieStreamKitServer({
    apiKey: config.apiKey,
    apiSecret: config.apiSecret,
  });

  return (req: Request & { rajutechie-streamkitUser?: Record<string, unknown> }, _res: Response, next: NextFunction) => {
    const authHeader = req.headers.authorization;
    if (!authHeader?.startsWith('Bearer ')) {
      return next();
    }

    try {
      const token = authHeader.slice(7);
      const decoded = server.verifyToken(token);
      req.rajutechie-streamkitUser = decoded as Record<string, unknown>;
      next();
    } catch {
      next();
    }
  };
}

export function requireRajutechieStreamKitAuth() {
  return (req: Request & { rajutechie-streamkitUser?: Record<string, unknown> }, res: Response, next: NextFunction) => {
    if (!req.rajutechie-streamkitUser) {
      res.status(401).json({ error: 'Authentication required' });
      return;
    }
    next();
  };
}
